### PR TITLE
add health checks for nodes in ingest and search

### PR DESCRIPTION
**Pull request by vangheem**
### Description
If no nodes are detected, the container should be considered unhealthy.

[ch-4624]

### How was this PR tested?
unit and manual

### CODE REVIEW

**Review by bloodbare — CHANGES_REQUESTED** [review]
Shouldnt be also to restart all pod in case one container fails?

File: nucliadb/nucliadb/ingest/consumer/pull.py [hunk 1]
```diff
@@ -192,6 +193,12 @@ async def finalize(self):
             await self.nc.close()
 
     async def subscription_worker(self, msg: Msg):
+        while len(NODES) == 0:
```
[the remark this answers is not in the document]
Comment (vangheem) [reply nucliadb/nucliadb/ingest/consumer/pull.py:196]: for example, if there is actually a lag on startup to get the nodes, I wouldn't want a constant exit and never being able to start up

File: nucliadb/nucliadb/ingest/consumer/pull.py [hunk 2]
```diff
@@ -192,6 +193,15 @@ async def finalize(self):
             await self.nc.close()
 
     async def subscription_worker(self, msg: Msg):
+        if len(NODES) == 0:
```
Comment (jotare) [inline nucliadb/nucliadb/ingest/consumer/pull.py:196]: Could you add a comment here. It's not obvious that we are waiting to get node metadata for chitchat with a 1s sleep.

Furthemore, why adding an extra ack_wait (10s) and wait only 1s? The remaining 9s wouldn't be enough?

Comment (vangheem) [reply nucliadb/nucliadb/ingest/consumer/pull.py]: yeah, this is silly. It should be a `while` and wait for it to become available.

The remaining 9s should be enough, however, the `in_progress` call resets the AckWait setting so it's like a new message until we're ready to work on it.

**Review by jotare — APPROVED** [review]

**Pull request merged**